analysis/src/CompletionBackEnd.ml

```diff
@@ -599,7 +599,7 @@ let completionsGetCompletionType ~full = function
   | _ -> None
 
 let rec getCompletionsForContextPath ~full ~opens ~rawOpens ~allFiles ~pos ~env
-    ~exact ~scope (contextPath : Completable.contextPath) =
+    ~exact ~scope ?(mode = `Regular) (contextPath : Completable.contextPath) =
```

Comment:
why poly variants here? ``` `Regular ```

Reply:
Pure laziness, I'll change it to a proper variant. Reason for having a mode overall is that I found a case where pipe completion breaks using my new approach, that I want to tackle in a separate PR/separate refactor.